src/shims/foreign_items.rs

```diff
@@ -433,7 +433,10 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 this.write_scalar(Scalar::from_f64(res), dest)?;
             }
 
-            // Target-specific shims
+            // Architecture-specific shims
+            "llvm.x86.sse2.pause" if this.tcx.sess.target.target.arch == "x86" || this.tcx.sess.target.target.arch == "x86_64" => {}
```

Comment:
@vakaras @divergentdave now that Miri has actual concurrency, do you think it would make sense to also yield the thread when this intrinsic is invoked?

Reply:
I think that it would make sense with out current scheduler.